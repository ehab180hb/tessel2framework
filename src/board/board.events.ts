import * as events from 'events'
export const boardEvents = new events()

export enum BoardEvents {
  UPDATE_BUTTONS = 'updateButtons',
  UPDATE_SCREEN = 'updateScreen',
}
