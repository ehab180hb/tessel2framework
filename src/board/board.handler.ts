import * as five from 'johnny-five'

const isTesting = process.env.NODE_ENV === 'test'

const Tessel = !isTesting ? require('tessel-io') : class {}

export const getBoard = () => {
  return new five.Board({
    io: new Tessel(),
  })
}
