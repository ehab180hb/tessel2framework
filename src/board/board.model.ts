export enum PIN {
  'a0',
  'a1',
  'a2',
  'a3',
  'a4',
  'a5',
  'a6',
  'a7',
  'b0',
  'b1',
  'b2',
  'b3',
  'b4',
  'b5',
  'b6',
  'b7',
}
