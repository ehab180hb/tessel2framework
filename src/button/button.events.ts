import { greenButton, redButton, blueButton } from '../button/button.handler'
import { BoardEvents, boardEvents } from '../board/board.events'

interface UpdateButtonsInput {
  greenButtonCb?: () => void
  redButtonCb?: () => void
  blueButtonCb?: () => void
}

export const attachButtonUpdate = (updateButtonsParams: UpdateButtonsInput): void => {
  const emptyButtonCb = () => {}
  const {
    greenButtonCb = emptyButtonCb,
    redButtonCb = emptyButtonCb,
    blueButtonCb = emptyButtonCb,
  } = updateButtonsParams
  greenButton(greenButtonCb)
  redButton(redButtonCb)
  blueButton(blueButtonCb)
}

export const updateButtons = (updateButtonsParams: UpdateButtonsInput) => {
  boardEvents.emit(BoardEvents.UPDATE_BUTTONS, updateButtonsParams)
}
