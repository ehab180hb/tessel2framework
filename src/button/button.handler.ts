import * as five from 'johnny-five'
import { PIN } from '../board/board.model'
import { ButtonsPinLayout } from '../config'
const { redButtonPin, greenButtonPin, blueButtonPin } = ButtonsPinLayout

const getButton = (pin: PIN) => (cb: () => void) => new five.Button(pin).on('press', cb)

export const redButton = getButton(redButtonPin)
export const greenButton = getButton(greenButtonPin)
export const blueButton = getButton(blueButtonPin)
