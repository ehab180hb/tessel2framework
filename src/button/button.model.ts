import * as five from 'johnny-five'
import { PIN } from '../board/board.model'

export interface ButtonCall {
  pin: PIN
  name: string
}

export interface DeviceButton {
  name: string
  pin: PIN
  listener: five.Button
}
