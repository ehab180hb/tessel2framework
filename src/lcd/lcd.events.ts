import { newMessage } from '../messages/messages.handler'
import { boardEvents, BoardEvents } from '../board/board.events'
import { fiveTypes } from '../typings/johnny-five'

interface UpdateScreenInput {
  row1?: string
  row2?: string
  lcd: fiveTypes.LCD
}

export const attachScreenUpdate = (updateScreenParams: UpdateScreenInput): void => {
  const emptyRow = ' '.repeat(16)
  const { row1 = emptyRow, row2 = emptyRow, lcd } = updateScreenParams
  const writeMsg = newMessage(lcd)
  writeMsg(0, row1)
  writeMsg(1, row2)
}

export const updateScreen = (updateScreenParams: UpdateScreenInput) => {
  boardEvents.emit(BoardEvents.UPDATE_SCREEN, updateScreenParams)
}
