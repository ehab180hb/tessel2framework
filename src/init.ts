import { boardEvents, BoardEvents } from './board/board.events'
import { attachScreenUpdate } from './lcd/lcd.events'
import { defaultButtons } from './config'
import { buttonHandler } from './button/button.handler'

export const initComponents = () => {
  boardEvents.on(BoardEvents.UPDATE_SCREEN, attachScreenUpdate)
  buttonHandler.initDefaultButtons(defaultButtons)
}
