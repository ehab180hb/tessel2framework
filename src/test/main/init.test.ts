import { attachScreenUpdate } from '../../lcd/lcd.events'
import { boardEvents, BoardEvents } from '../../board/board.events'
import { initComponents } from '../../init'
import { initDefaultButtons } from '../../button/button.handler'

describe('init tests', () => {
  beforeEach(() => {
    ;(attachScreenUpdate as any) = jest.fn()
    ;(initDefaultButtons as any) = jest.fn()
  })

  test('Screen update methods attacher is called', () => {
    initComponents()
    boardEvents.emit(BoardEvents.UPDATE_SCREEN, 'x')
    expect(attachScreenUpdate).toBeCalledWith('x')
  })

  test('Button update methods attacher is called', () => {
    initComponents()
    expect(initDefaultButtons).toMatchSnapshot()
  })
})
