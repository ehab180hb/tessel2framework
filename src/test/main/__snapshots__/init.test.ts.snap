// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init tests Button update methods attacher is called 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "name": "red",
          "pin": 13,
        },
        Object {
          "name": "green",
          "pin": 14,
        },
        Object {
          "name": "blue",
          "pin": 15,
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
